graph/bidirectional_dijkstra.cpp

```diff
@@ -0,0 +1,259 @@
+/**
+ * @file
+ * @brief [Bidirectional Dijkstra Shortest Path Algorithm]
+ * (https://www.coursera.org/learn/algorithms-on-graphs/lecture/7ml18/bidirectional-dijkstra)
+ *
+ * @author [Marinovksy](http://github.com/Marinovsky)
+ *
+ * @details
+ * This is basically the same Dijkstra Algorithm but faster because it goes from
+ * the source to the target and from target to the source and stops when
+ * finding a vertex visited already by the direct search or the reverse one.
+ * Here some simulations of it:
+ * https://www.youtube.com/watch?v=DINCL5cd_w0&t=24s
+ */
+
+#include <math.h>
+#include <cassert>
+#include <cstdio>
+#include <iostream>
+#include <limits>
+#include <set>
+#include <unordered_map>
+#include <queue>
+#include <utility>
+#include <vector>
```

Comment:
Add a brief description of what the library/header adds (see the example below).

        #include <assert>     /// for assert
        #include <iostream>  /// for io operations